### PR TITLE
Add support for openSUSE

### LINKED ISSUE #39: Add support for openSUSE
URL: `https://connect.opensuse.org/pg/profile/{username}`

### PR DESCRIPTION
**Pull request by biscuitsnake**
Closes #39

### CODE REVIEW

**Comment by biscuitsnake** [comment]
Also made the openhub icon grey, hope that's okay :)

File: username_api.py [hunk 1]
```diff
@@ -161,4 +164,4 @@ def check(website, username):
 	return jsonify(check_username(website, username))
 
 if __name__ == '__main__':
-	app.run(host='0.0.0.0', port=8521)
\ No newline at end of file
+	app.run(host='0.0.0.0', port=8521)
```
Comment (jayvdb) [inline username_api.py]: unrelated change

File: tests/test_get_avatar.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 def load_test_cases(type):
 	res = []
 	for website in data:
-		if website in ['behance', 'openhub']:
+		if website in ['behance', 'openhub', 'opensuse']:
```
Comment (jayvdb) [inline tests/test_get_avatar.py]: this could be derived from the yaml I think.

File: username_api.py [hunk 3]
```diff
@@ -73,10 +73,13 @@ def check_username(website, username):
 			'possible': possible,
 		}
 
-	if website in ['pinterest', 'gitlab']:
+	if website in ['pinterest', 'gitlab', 'opensuse']:
```
Comment (jayvdb) [inline username_api.py]: this can be derived from the yaml; `website in patterns['phrase']`

File: websites.yml [hunk 4]
```diff
@@ -68,10 +73,15 @@ avatar:
...
   pinterest:
     key: image_xlarge_url
   soundcloud: opengraph
   tumblr: opengraph
   twitter:
     url: https://twitter.com/{u}/profile_image?size=original
+phrase:
+  opensuse: "About me"
+  pinterest: "{u}"
+  gitlab: "@{u}"
```
Comment (manu-chroma) [inline websites.yml]: I think we can have `{u}` for gitlab as well. No need for `@`

Also, to make it clearer, I'd prefer having `{username}` here.

Comment (manu-chroma) [reply websites.yml]: another thought, shouldn't we put phrase key under these websites rather than creating a separate section for it. 
cc: @jayvdb

File: static/css/main.css [hunk 5]
```diff
@@ -81,6 +81,16 @@ html,body
...
+  filter: brightness(70%) contrast(200%) grayscale(90%);
+}
+
+.fa-opensuse {
+  display: inline-block;
+  width: 36px;
+  height: 36px;
+  background-image: url("https://en.opensuse.org/images/9/92/Geeko-button.png");
+  background-size: contain;
+  filter: grayscale(90%);
```
Comment (manu-chroma) [inline static/css/main.css]: I like what you've done with this. It looks good. Try making the black colour darker, if possible

File: websites.yml (hunk 4 above)
Comment (biscuitsnake) [reply websites.yml]: I put the @ in because {u} didn't work on it's own. phrase is separate because the way the other sections are named, putting a phrase in them would be confusing, but up to @jayvdb to decide.

Comment (jayvdb) [reply websites.yml]: If we put it under `urls`, then they would need to change structure from `<website>: <url>` to
```
urls:
  <website>:
    url: <url>
    confirmation_phrase: {u}
```

Not sure about that.

Comment (manu-chroma) [reply websites.yml]: makes sense. we can preserve this one then

Comment (jayvdb) [reply websites.yml]: We could do something funky like use a hash fragment to store the string ; i.e. http://www.gerv.net/software/fragment-search/
e.g.
`<website>: <url>#!s!{u}`

Comment (jayvdb) [reply websites.yml]: If we keep this section, which IMO is good enough, then it should be moved under `urls` as it is directly related to the username urls for the websites.

It should also be renamed to be more explicit.  I suggest something like `content_verification` or `HTTP404_emulation_phrase`

Comment (biscuitsnake) [reply websites.yml]: @manu-chroma surely {u} is clearer than {username} as that's what is being used in the urls section? just my opinion, I don't know

Comment (manu-chroma) [reply websites.yml]: It is obvious in the urls. I thought it might not be when coming to the ``content_verification`` section. I don't mind having ``{u}`` :smiley:

Comment (biscuitsnake) [reply websites.yml]: @manu-chroma okay, I moved the section underneath urls so it should be fine,
if that's all issues resolved, can you merge?

Comment (manu-chroma) [reply websites.yml]: > We could do something funky like use a hash fragment to store the string ; i.e. http://www.gerv.net/software/fragment-search
e.g.
<website>: <url>#!s!{u}

will read up on this :P

**Comment by manu-chroma** [comment]
@jayvdb pls review

File: websites.yml [hunk 6]
```diff
@@ -68,10 +77,11 @@ avatar:
...
   instagram: opengraph
+  opensuse: false
   openhub: false
   pinterest:
     key: image_xlarge_url
   soundcloud: opengraph
   tumblr: opengraph
   twitter:
-    url: https://twitter.com/{u}/profile_image?size=original
+    url: https://twitter.com/{u}/profile_image?size=original
```
Comment (jayvdb) [inline websites.yml]: what is going on here!

Comment (biscuitsnake) [reply websites.yml]: whoops, fixed

Comment (biscuitsnake) [reply websites.yml]: @jayvdb

File: websites.yml [hunk 7]
```diff
@@ -6,6 +6,11 @@ urls:
   pinterest: https://in.{w}.com/{u}
   tumblr: https://{u}.{w}.com
   facebook: https://mbasic.{w}.com/{u}
+  opensuse: https://connect.{w}.org/pg/profile/{u}
+content_verification:
+  opensuse: "About me"
+  pinterest: "{u}"
+  gitlab: "@{u}"
```
Comment (jayvdb) [inline websites.yml]: alpha order please

File: websites.yml [hunk 8]
```diff
@@ -48,6 +53,10 @@ username_patterns:
     characters: a-zA-Z0-9_-
     min_length: 1
     max_length: 25
+  opensuse:
```
Comment (jayvdb) [inline websites.yml]: alpha order please

File: websites.yml [hunk 9]
```diff
@@ -68,6 +77,7 @@ avatar:
   github: opengraph
   gitlab: opengraph
   instagram: opengraph
+  opensuse: false
```
Comment (jayvdb) [inline websites.yml]: alpha order please

File: websites.yml (hunk 8 above)
Comment (biscuitsnake) [reply websites.yml]: @jayvdb reordered all

**Comment by biscuitsnake** [comment]
\o/
